pandas/_libs/period.pyx

```diff
@@ -718,9 +724,8 @@ cdef class _Period(object):
         if ordinal == iNaT:
             return NaT
         else:
-            self = _Period.__new__(cls)
-            self.ordinal = ordinal
-            self.freq = cls._maybe_convert_freq(freq)
+            freq = cls._maybe_convert_freq(freq)
+            self = _Period.__new__(cls, ordinal, freq)
```

Comment:
I don't think so.  `cls._from_ordinal` gets called at the end of `Period.__new__`, so I expect it explicitly needs to be `_Period`.